community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -126,7 +126,7 @@ TransactionType upgradeToSchemaTransaction() throws InvalidTransactionTypeKernel
     // whereas others, such as timestamp or txId when transaction starts, even locks, needs to be set in #initialize().
     private TransactionState txState;
     private LegacyIndexTransactionState legacyIndexTransactionState;
-    private TransactionType transactionType; // Tracks current state of transaction, which will upgrade to WRITE or SCHEMA mode when necessary
+    private TransactionWriteState writeState; // Tracks current state of transaction, which will upgrade to WRITE or SCHEMA mode when necessary
```

Comment:
Please transform this comment with some more details into class level javadoc for `TransactionWriteState` class